src/main/java/services/ParseSchoolSubjects.java

```diff
@@ -0,0 +1,83 @@
+package services;
+
+import org.jetbrains.annotations.NotNull;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ParseSchoolSubjects {
+    private static Logger logger =
+            LoggerFactory.getLogger("services.ParseSection");
+
+    public static @NotNull String parseSchoolSubject(String rawData) {
+        logger.info("parsing raw school data...");
+        Document doc = Jsoup.parse(rawData);
+        Elements scripts = doc.select("script");
+        String output = "";
+        for(Element script : scripts) {
+            String scriptHTML = script.html();
+            if(scriptHTML.contains("classSearch")) {
+                output = scriptHTML.substring(
+                        scriptHTML.indexOf('{') + 1, scriptHTML.lastIndexOf('}'));
+                output = output.substring(
+                        output.indexOf('{'), output.lastIndexOf('}') + 1
+                );
+            }
+        }
+        return output;
+    }
+
+    public static Map<String, String> parseSchool(String rawData) {
+        String HTML = parseSchoolSubject(rawData);
+        List<String> schools = new ArrayList<>();
+        List<String> descriptions = new ArrayList<>();
+        Map<String, String> map = new HashMap<>();
+        String output = HTML.substring(HTML.lastIndexOf("acad_groups"), HTML.indexOf("nyuSchoolsUrl"));
+        output = output.substring(output.indexOf("[") + 1, output.lastIndexOf("]"));
+        String[] outputs = output.split(",");
+        for(String value : outputs) {
+            if(value.contains("acad_group")) {
```

Comment:
We could probably improve perf by combining this forloop with the one on line 55. It's not that important though, I can accept without a fix.